File: striver/S3L1/sort_0_1_2.py
class Solution(object):
    def sortColors(self, nums):
        c0=int(0)
        c1=int(0)
        c2=int(0)
        for i in range(0,len(nums)):
            if nums[i] == 0 :
                c0+=1
            elif nums[i] == 1:
                c1+=1
            else:
                c2+=1
        i = int(0)
        while (c0 > 0):
            nums[i]=0
            c0-=1
            i+=1
        while (c1 > 0):
            nums[i]=1
            c1-=1
            i+=1
        while (c2 > 0):
            nums[i]=2
            c2-=1
            i+=1

File: striver/S3L1/sort_0_1_2.java
package S3L1;

public class sort_0_1_2 {
    public void sortColors(int[] nums) {
        int c0 = 0;
        int c1 = 0;
        int c2 = 0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==0)c0++;
            else if(nums[i]==1)c1++;
            else c2++;
        }
        int i=0;
        while(c0 != 0){
            nums[i++]=0;
            c0--;
        }
        while(c1 != 0){
            nums[i++]=1;
            c1--;
        }
        while(c2 != 0){
            nums[i++]=2;
            c2--;
        }
    }
    public void sortColors(int[] nums) {
      int low = 0;
      int mid = 0;
      int high = nums.length-1;
      while(mid <= high){
        if(nums[mid] == 0) {
            int temp = nums[low];
            nums[low] = nums[mid];
            nums[mid] = temp;
            low++;
            mid++;
        } else if(nums[mid] == 1){
            mid++;
        }else {
            int temp = nums[high];
            nums[high] = nums[mid];
            nums[mid] = temp;
            high--;
        }
      }
    }
}